src/main/java/controller/Controller.java

```diff
@@ -0,0 +1,215 @@
+package controller;
+
+import model.*;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+/**
+ * Created by Jef1771 on 7/15/17.
+ */
+public class Controller {
+
+    //DAO's
+    PlayerDAO playerDAO;
+    AreaDAO areaDAO;
+    EnemyDAO enemyDAO;
+    WeaponDAO weaponDAO;
+    TreasureDAO treasureDAO;
+
+
+    Map<Integer, Enemy> enemies;
+    Area area;
+    Player currentPlayer;
+
+    public Controller(){
+        playerDAO = new PlayerDAO();
+        areaDAO = new AreaDAO();
+        enemyDAO = new EnemyDAO();
+        weaponDAO = new WeaponDAO();
+        treasureDAO = new TreasureDAO();
+    }
+
+    public void setCurrentPlayer(String name){
+        currentPlayer = playerDAO.getPlayer(name);
+        area = currentPlayer.currentArea;
+        populateEnemies();
+    }
+
+    public void setCurrentArea(Area a){
+        area = a;
+        currentPlayer.currentArea = a;
+        populateEnemies();
+    }
+
+    public void createNewPlayer(String name){
+        currentPlayer = playerDAO.createPlayer(name);
+        area = currentPlayer.currentArea;
+        populateEnemies();
+    }
+
+    //VIEW USE THESE METHODS TO GET INFO YOU NEED
+
+    /**
+     * Get a list of all the player names in the DB
+     *
+     * @return List of all player names
+     */
+    public List<String> getAllPlayers(){
+        return playerDAO.getNamesOfAllPlayers();
+    }
+
+    /**
+     * Get a Map of all the enemies names in the current Area
+     *
+     *
+     *
+     * @return List of all enemies names
+     */
+    public Map<Integer, String> getAllEnemies(){
```

Comment:
We switched enemy back to a Map<Enemy, Integer> with the count so I'll return to you a map of <Integer, Enemy> so you can still give me the ID of the enemy that the player attacks and you can also have the enemy object